src/DFAMin.hs

```diff
@@ -35,6 +34,26 @@ import qualified Data.List as List
 --      end;
 -- end;
 
+-- observation : Q is always subset of P
+-- let R = P \ Q. then following algorithm is the equivalent of the Hopcroft's Algorithm
```

Comment:
I really like the idea of deduplicating the state like this :) But the equivalence of the algorithms is not readily apparent to me (other than the initial state).

I will continue to consult the Wikipedia article on this, but if you had a proof sketch to provide, that would be nice.

Reply:
`R = P \ Q` is held through the execution of algorithm.
i.e. `R` and `Q` are disjoint and `P = R ∪ Q`.

So, while the original algorithm iterate over `P` to choose a set `Y`, the new algorithm iterate over both `R` and `Q`.
The good part is that we don't need to check whether `Y ∈ Q` or not since `Y ∈ R` means `Y ∈ P \ Q` and `Y ∈ Q` means `Y ∈ P ∩ Q`.